### PR TITLE
Support for FIFO SNS Topics

### PR DESCRIPTION
**Pull request by azec-pdx**
## what
* Adds ability to support deploying FIFO - type SNS Topics

## why
* Customers might need FIFO SNS Topics with FIFO SQS queues subscribed
* This allows them to use module in following way
   ```terraform
    module "sns" {
        source = "git::https://github.com/cloudposse/terraform-aws-sns-topic.git//?ref=tags/<FUTURE_TAG>"
        
        fifo_topic                  = var.fifo_topic
        name                        = module.this.id # e.g. mytopic.fifo   
        regex_replace_chars         = var.regex_replace_chars     # e.g. "/[^.a-zA-Z0-9-]/"  - required to prevent removal of '.' char in '.fifo' suffix
        sqs_dlq_enabled             = false # Until valid case neeed for this and until CP fixes https://github.com/cloudposse/terraform-aws-sns-topic/blob/master/main.tf#L20
        subscribers                 = var.subscribers
        content_based_deduplication = var.content_based_deduplication
      
        # SNS Topic policies related variables
        sns_topic_policy_json                  = var.sns_topic_policy_json
        allowed_aws_services_for_sns_published = var.allowed_aws_services_for_sns_published
        allowed_iam_arns_for_sns_publish       = var.allowed_iam_arns_for_sns_publish
    }
   ```
   and then upstream to use this as: 
   ```terraform
    module "my_topic_and_subscribers" {
      source = "../../components/sns_topic_w_subscribers"
    
      environment                 = var.environment
      fifo_topic                  = true # has to be fifo for FIFO SQS queue subscribers (which all are)
      content_based_deduplication = true
      name                        = "my-topic.fifo" # suffix .fifo is required by AWS for this type of SNS Topic
      namespace                   = var.namespace
      regex_replace_chars         = "/[^.a-zA-Z0-9-]/" # required to prevent removal of '.' char in '.fifo' suffix
      region                      = var.region
      subscribers = {
        my_sqs_fifo = {
          endpoint               = module.my_queue_fifo.queue_arn # ARN of the SQS that will be subscribed to SNS topic
          endpoint_auto_confirms = true
          protocol               = "sqs",
          raw_message_delivery   = false
        }
      }
      sns_topic_policy_json = data.aws_iam_policy_document.my_topic_policy_document.json
    }

## references
* N/A

### CODE REVIEW

File: examples/complete/variables.tf [hunk 1]
```diff
@@ -2,3 +2,9 @@ variable "region" {
   type        = string
   description = "AWS region"
 }
+
+variable "allowed_aws_services_for_sns_published" {
+  type        = list(string)
+  description = "AWS services that will have permission to publish to SNS topic. Used when no external json policy is used."
```
Comment (aknysh) [inline examples/complete/variables.tf:8]:
```suggestion
  description = "AWS services that will have permission to publish to SNS topic. Used when no external JSON policy is used"
```

File: variables.tf [hunk 2]
```diff
@@ -73,3 +73,15 @@ variable "delivery_policy" {
...
+
+variable "fifo_topic" {
+  type        = bool
+  description = "(Optional) Boolean indicating whether or not to create a FIFO (first-in-first-out) topic"
+  default     = false
+}
+
+variable "content_based_deduplication" {
+  type        = bool
+  description = "(Optional) Enables content-based deduplication for FIFO topics"
```
Comment (aknysh) [inline variables.tf:85]:
```suggestion
  description = "Enable content-based deduplication for FIFO topics"
```

File: variables.tf [hunk 3]
```diff
@@ -73,3 +73,15 @@ variable "delivery_policy" {
   description = "The SNS delivery policy as JSON."
   default     = null
 }
+
+variable "fifo_topic" {
+  type        = bool
+  description = "(Optional) Boolean indicating whether or not to create a FIFO (first-in-first-out) topic"
```
Comment (aknysh) [inline variables.tf:79]:
```suggestion
  description = "Whether or not to create a FIFO (first-in-first-out) topic"
```

**Review by aknysh — COMMENTED** [review]
thanks @azec-pdx  please see comments

**Comment by azec-pdx** [comment]
> thanks @azec-pdx please see comments

Hey @aknysh , I have addressed your comments....

**Review by aknysh — APPROVED** [review]
thanks @azec-pdx

**Pull request merged**